### ISSUE TITLE
Wait for flag's callback be called when async

### ISSUE THREAD

**Issue by vohof** [scottcorgan/nash#42]
```
'use strict'

var server = require('./lib/server.js').server
var port = process.env.PORT || 8080
var fs = require('fs')
var cli = require('nash')()

cli
  .flag('-c', '--config').async().handler(function(file, done) {
    console.log(file)
    fs.readFile(file, function(err, content) {
      if (err) throw err
      cli.set('config', JSON.parse(content))
      done()
    })
  })

cli
  .command('run').handler(run)


cli.run(process.argv)

function run() {
  server.listen(port, function() {
    console.log(`listening on ${port}.`)
    // setTimeout(function() {
      console.log(cli.get('config'))
    // }, 1000)
  })
}
```

config will be `undefined` :sob: It doesn't seem to wait for the async flag's handler to finish before running the command. I must be doing something wrong.

```
package.json
listening on 8080.
undefined
```

**Comment by scottcorgan** [comment]
I have opened a discussion on whether or not we should remove the async option and just assume async on all operations.

I'd be interested in hearing your thoughts.

I'll fix this bug in the meantime.

**Comment by scottcorgan** [comment]
Closing this as this will be fixed in #39
